File: finanze/application/ports/credentials_port.py
import abc
from datetime import datetime
from typing import Optional
from uuid import UUID

from domain.native_entity import EntityCredentials, FinancialEntityCredentialsEntry


class CredentialsPort(metaclass=abc.ABCMeta):
    async def get(self, entity_id: UUID) -> Optional[EntityCredentials]:
        raise NotImplementedError

    async def get_available_entities(self) -> list[FinancialEntityCredentialsEntry]:
        raise NotImplementedError

    async def save(self, entity_id: UUID, credentials: EntityCredentials):
        raise NotImplementedError

    async def delete(self, entity_id: UUID):
        raise NotImplementedError

    async def update_last_usage(self, entity_id: UUID):
        raise NotImplementedError

    async def update_expiration(self, entity_id: UUID, expiration: Optional[datetime]):
        raise NotImplementedError

File: mobile/src/application/ports/core/credentialsPort.ts
import { EntityCredentials, FinancialEntityCredentialsEntry } from "@/domain"

export interface CredentialsPort {
  get(entityId: string): Promise<EntityCredentials | null>
  getAvailableEntities(): Promise<FinancialEntityCredentialsEntry[]>
  save(entityId: string, credentials: EntityCredentials): Promise<void>
  delete(entityId: string): Promise<void>
  updateLastUsage(entityId: string): Promise<void>
  updateExpiration(entityId: string, expiration: string | null): Promise<void>
}
